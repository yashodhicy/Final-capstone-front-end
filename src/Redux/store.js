import { configureStore } from '@reduxjs/toolkit';
import HouseSlice from './HouseSlice';
import reducer from './reservation/reducer';


const store = configureStore({
  reducer: {
    Houses: HouseSlice,
    reservations: reducer
  },
});

export default store;
