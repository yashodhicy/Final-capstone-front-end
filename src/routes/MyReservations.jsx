import React, { useEffect } from "react";
import ReservationItem from "../components/Reservations/ReservationItem";
import { useDispatch, useSelector } from "react-redux";
import { getReservations } from "../Redux/reservation/middlewares";

const MyReservations = () => {
  const dispatch = useDispatch();
  const reservations = useSelector((state) => state.reservations);
  useEffect(() => {
    dispatch(getReservations());
  }, [dispatch]);
  return (
    <article className="p-3 d-flex flex-column align-items-center w-100">
      <h1 className="text-center text-secondary mb-5">My reservations</h1>
      <ul className="p-0 w-100" style={{ maxWidth: 1500 }}>
        {reservations.reservations.map((reservation, index) => (
          <ReservationItem key={index} reservation={reservation} />
        ))}
      </ul>
    </article>
  );
};

export default MyReservations;
