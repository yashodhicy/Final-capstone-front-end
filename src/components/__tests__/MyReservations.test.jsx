/* eslint-env jest */
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../Redux/store';
import MyReservations from '../../routes/MyReservations'

describe('Test MyReseravation  Component', () => {
    test ('it renders correctly', () => {
        const tree = renderer
        .create(
            <Provider store={store}>
                <Router>
                    <MyReservations />
                </Router>
            </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
});
