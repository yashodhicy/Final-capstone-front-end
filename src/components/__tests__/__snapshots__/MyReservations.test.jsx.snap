// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MyReseravation  Component it renders correctly 1`] = `
<article
  className="p-3 d-flex flex-column align-items-center w-100"
>
  <h1
    className="text-center text-secondary mb-5"
  >
    My reservations
  </h1>
  <ul
    className="p-0 w-100"
    style={
      {
        "maxWidth": 1500,
      }
    }
  />
</article>
`;
